Demo/WhisperDemo/WhisperDemo/ViewController.swift

```diff
@@ -23,28 +23,75 @@ class ViewController: UIViewController {
     return label
     }()
 
-  lazy var presentButton: UIButton = {
+  lazy var presentButton: UIButton = { [unowned self] in
     let button = UIButton()
     button.addTarget(self, action: "presentButtonDidPress:", forControlEvents: .TouchUpInside)
-    button.setTitle("Present", forState: .Normal)
-    button.setTitleColor(UIColor.blackColor(), forState: .Normal)
-    button.layer.borderColor = UIColor.blackColor().CGColor
-    button.layer.borderWidth = 1.5
-    button.layer.cornerRadius = 7.5
+    button.setTitle("Present and silent", forState: .Normal)
 
     return button
     }()
 
-  var shouldChange = false
+  lazy var showButton: UIButton = { [unowned self] in
+    let button = UIButton()
+    button.addTarget(self, action: "showButtonDidPress:", forControlEvents: .TouchUpInside)
+    button.setTitle("Show", forState: .Normal)
+
+    return button
+    }()
+
+  lazy var presentPermanentButton: UIButton = { [unowned self] in
+    let button = UIButton()
+    button.addTarget(self, action: "presentPermanentButtonDidPress:", forControlEvents: .TouchUpInside)
+    button.setTitle("Present permanent Whisper", forState: .Normal)
+
+    return button
+    }()
+
+  lazy var newControllerButton: UIBarButtonItem = { [unowned self] in
+    let button = UIBarButtonItem()
+    button.title = "Next"
+    button.target = self
+    button.action = "nextButtonDidPress"
+
+    return button
+    }()
+
+  lazy var containerView: UIView = {
+    let view = UIView()
+    view.backgroundColor = UIColor.grayColor()
+
+    return view
+    }()
 
   override func viewDidLoad() {
     super.viewDidLoad()
 
-    title = "Whisper".uppercaseString
     view.backgroundColor = UIColor.whiteColor()
 
+    navigationItem.rightBarButtonItem = newControllerButton
+    navigationItem.backBarButtonItem = UIBarButtonItem(title: nil, style: .Plain, target: nil, action: nil)
+
     view.addSubview(scrollView)
-    for subview in [icon, titleLabel, presentButton] { scrollView.addSubview(subview) }
+    for subview in [icon, titleLabel, presentButton, showButton, presentPermanentButton] { scrollView.addSubview(subview) }
+
+    for button in [presentButton, showButton, presentPermanentButton] {
+      button.setTitleColor(UIColor.grayColor(), forState: .Normal)
+      button.layer.borderColor = UIColor.grayColor().CGColor
+      button.layer.borderWidth = 1.5
+      button.layer.cornerRadius = 7.5
+    }
+
+    guard let navigationController = navigationController else { return }
```

Comment:
@RamonGilabert @zenangst I don't like the early return in the middle of the func, still feel that guard is our next `map` that we're using in the wrong way. It should be used for error handling, to throw errors, or for early return in the beginning of the functions. What do you guys think?